hearth/lib/hearth/interceptor_list.rb

```diff
@@ -1,38 +1,35 @@
 # frozen_string_literal: true
 
 module Hearth
-  # A list of Interceptors.
+  # A list of {Interceptor}s or classes that respond to Interceptor
+  # hook methods.
   class InterceptorList
     include Enumerable
 
-    # Initialize an InterceptorList.
-    #
-    # @param [Array] interceptors ([]) A list of interceptors.
+    # @param [Array<Interceptor>] interceptors ([])
     def initialize(interceptors = [])
-      unless interceptors.respond_to?(:each)
-        raise ArgumentError, 'Interceptors must be an enumerable'
-      end
-
       @interceptors = []
-      interceptors.each { |i| add(i) }
+      interceptors.each { |i| append(i) }
     end
 
-    # Add an interceptor.
-    #
-    # @param [Object] interceptor
-    def add(interceptor)
-      if interceptor.respond_to?(:each)
-        interceptor.each { |i| add(i) }
-      else
-        unless valid_interceptor?(interceptor)
-          raise ArgumentError,
-                'Invalid Interceptor - must implement at least one hook method.'
-        end
-
-        @interceptors << interceptor
+    # @param [Interceptor] interceptor
+    def append(interceptor)
+      unless valid_interceptor?(interceptor)
+        raise ArgumentError,
+              'Invalid interceptor - must respond to any of: ' \
+              "#{Interceptor.hooks.join(', ')}"
       end
+
+      @interceptors << interceptor
+    end
+    alias << append
+
+    # @param [InterceptorList] other
+    # @return [InterceptorList] self
+    def concat(other)
```

Comment:
Why this change? To make it behave more like Ruby lists where you can't use `<<` to concat another list?

Reply:
Yeah this follows the enumerable interface.